morango/utils/uuids.py

```diff
@@ -110,9 +110,12 @@ class DatabaseManager(models.Manager):
 
     def create(self, **kwargs):
 
-        # set current flag to false for all database_id models
-        DatabaseIDModel.objects.update(current=False)
-        super(DatabaseManager, self).create(**kwargs)
+        # do within transaction so we always have a current database ID
+        with transaction.atomic():
```

Comment:
Were you able to confirm that this kind of thing does guarantee atomicity? E.g. by putting pdb breaks in between, and then opening an additional shell in a separate process and connecting to the DB to check? Or using sqlitebrowser?